src/modules/currency/useCurrency.ts

```diff
@@ -0,0 +1,16 @@
+import { useStore } from '@store-unit/react';
+import { useCallback } from 'react';
+import { preferenceStore } from 'src/ui/features/appearance';
+
+export function useCurrency() {
+  const { currency } = useStore(preferenceStore);
+  return {
+    currency,
+    setCurrency: useCallback((value: string) => {
```

Comment:
I don't think it's worth to create this function for every hook call. It's used only once: on the config page. We can access `preferenceStore` directly there